### PR TITLE
Ruby Course - Pattern Matching: Improve Lesson Readability

### PR DESCRIPTION
**Pull request by tyler-hawkins**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

The lack of commas, run-on sentences, and overall sentence structure in a lot of places makes this lesson hard/disengaging to read. This PR attempts to improve on that, at least a little bit.

**2. This PR:**

* Removes some punctuation and capitalizes sections where appropriate to adhere to the layout style guide. 
* Adds punctuation and changes some wording/capitalization where necessary to prevent run-on sentences and allow for easier readability/flow.

**3. Additional Information:**
Just a small disclaimer: as a TOP student, I reached this lesson without having the prior knowledge of the subject to confidently reword some portions that stuck out to me. I would urge somebody with the knowledge of this lesson to double-check that the edits made in this PR still make sense and convey the right information in its new context. 
For instance, there's an entire repeated paragraph & accompanying code block: 
> "The case/in format is best used when..." 

in the lesson and I can't say with 100% certainty whether it belongs there or not. 😅

### CODE REVIEW

File: ruby/advanced_ruby/pattern_matching.md [hunk 1]
```diff
@@ -592,7 +592,7 @@ end
 # => [7, 8, 9]
 ~~~
 
-As a last example we'll consider a common use case which will hopefully show where the find pattern could be a better fit than a conventional Ruby solution. It's not uncommon in Ruby to find yourself with an array of hashes or json data, and you might need to locate a record from that data. You need to match that data on a few hash keys. The data might look something like
+As a last example, we'll consider a common use case which will hopefully show where the find pattern could be a better fit than a conventional Ruby solution. It's not uncommon in Ruby to find yourself with an array of hashes or JSON data, and you might need to locate a record from that data. You need to match that data on a few hash keys. The data might look something this.
```
Comment (CouchofTomato) [inline ruby/advanced_ruby/pattern_matching.md:595]: The last sentence here doesn't read well. Maybe a word was missed out?

**Review by CouchofTomato — CHANGES_REQUESTED** [review]
Great work. Just a small edit please.

File: ruby/advanced_ruby/pattern_matching.md (hunk 1 above)
Comment (tyler-hawkins) [reply ruby/advanced_ruby/pattern_matching.md:595]: Yeah, you're totally right, I must have missed it! Thank you for spotting that.

**Review by CouchofTomato — APPROVED** [review]

**Comment by CouchofTomato** [comment]
@tyler-hawkins thanks. This is a great improvement.

**Pull request merged**